# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RendererScreenshotGrabber.cpp
# Commit: f3b490c076d6
# Full Hash: f3b490c076d623d6ef3bcfd297a0522d9727477c
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2020-01-23 21:42:24
# Regressor Bug: 1602643
# File Overlap Count: 1
# Description:
#   Bug 1602643 - Disable WebRender compositor dinamically for async screenshot r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D59658
# ==============================================================================

diff -r 0c5fb3a8a94d -r f3b490c076d6 gfx/webrender_bindings/RendererScreenshotGrabber.cpp
--- a/gfx/webrender_bindings/RendererScreenshotGrabber.cpp	Thu Jan 23 15:35:45 2020 +0200
+++ b/gfx/webrender_bindings/RendererScreenshotGrabber.cpp	Thu Jan 23 04:37:28 2020 +0000
@@ -16,27 +16,33 @@
 }
 
 void RendererScreenshotGrabber::MaybeGrabScreenshot(
-    Renderer* aRenderer, const gfx::IntSize& aWindowSize) {
-  if (ProfilerScreenshots::IsEnabled()) {
+    RendererOGL* aRendererOGL, const gfx::IntSize& aWindowSize) {
+  bool isEnabled =
+      ProfilerScreenshots::IsEnabled() && aRendererOGL->EnsureAsyncScreenshot();
+
+  if (isEnabled) {
     if (!mProfilerScreenshots) {
       mProfilerScreenshots = new ProfilerScreenshots();
     }
 
-    GrabScreenshot(aRenderer, aWindowSize);
+    GrabScreenshot(aRendererOGL->GetRenderer(), aWindowSize);
   } else if (mProfilerScreenshots) {
-    Destroy(aRenderer);
+    Destroy(aRendererOGL->GetRenderer());
   }
 }
 
-void RendererScreenshotGrabber::MaybeProcessQueue(Renderer* aRenderer) {
-  if (ProfilerScreenshots::IsEnabled()) {
+void RendererScreenshotGrabber::MaybeProcessQueue(RendererOGL* aRendererOGL) {
+  bool isEnabled =
+      ProfilerScreenshots::IsEnabled() && aRendererOGL->EnsureAsyncScreenshot();
+
+  if (isEnabled) {
     if (!mProfilerScreenshots) {
       mProfilerScreenshots = new ProfilerScreenshots();
     }
 
-    ProcessQueue(aRenderer);
+    ProcessQueue(aRendererOGL->GetRenderer());
   } else if (mProfilerScreenshots) {
-    Destroy(aRenderer);
+    Destroy(aRendererOGL->GetRenderer());
   }
 }
 